cf/crypto.m4

```diff
@@ -134,6 +134,14 @@ if test "$with_openssl" != "no"; then
         LDFLAGS="${saved_LDFLAGS}"
 fi
 
+if test "$openssl" = "yes"; then
+    AC_CHECK_LIB([crypto],
+                 [OSSL_EC_curve_nid2name],
+                 [AC_DEFINE_UNQUOTED([HAVE_OPENSSL_30], 1,
+                                     [whether OpenSSL is 3.0 or higher])]
+                 )
+fi
+
```

Comment:
Just compile a C program with `#if OPENSSL_VERSION_PREREQ(3,0)`.

Reply:
We need a HAVE_OPENSSL_30 for places where we can't include OpenSSL headers.  Yes, a better autoconf macro would be nice.